File: ecommerce-fe/src/app/models/orderedProducts.ts
import {PurchasedHistoryModel} from './purchasedHistoryModel';
import {Product} from './product';

export interface OrderedProducts{
  quantityProduct: number;
  idProduct: number;
  idPurchasedHistory: number;
  purchasedHistory: PurchasedHistoryModel;
  product: Product;
}

File: E-commerce Application BE/E-commerce Application/Models/OrderedProducts.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebApplication1.Models;

[Table("ordered_products")]
public class OrderedProducts
{
    [Column("quantityProduct")]
    public int QuantityProduct { get; set; }
    
    [ForeignKey(nameof(Product))]
    public int idProduct { get; set; }
    public Product Product { get; set; }

    [ForeignKey(nameof(PurchasedHistory))]
    public int idPurchasedHistory { get; set; }
    public PurchasedHistory PurchasedHistory { get; set; }
    
}